### ISSUE TITLE
Coloring and Theming - Add closure doc

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell.github.io#1503]
From [this Discord comment](https://discord.com/channels/601130461678272522/614593951969574961/1268892496977723464) from @sholderbach:

> his never made it into the proper documentation but if the condition is only dependent on the individual value you can use: https://www.nushell.sh/blog/2022-12-20-nushell-0.73.html#more-sophisticated-coloring-of-data-values-using-closures-webbedspace

> Worth updating [Coloring and Theming in Nu](https://www.nushell.sh/book/coloring_and_theming.html#primitive-values)

**Comment by Kissaki** [comment]
Does this still work? It doesn't seem to work for me.

```nu
# typing `true` in inline editor becomes cyan

$env.config.color_config.shape_bool = 'green'
# typing `true` in inline editor becomes green

$env.config.color_config.shape_bool = {|x| 'green' }
# typing `true` in inline editor becomes cyan
```

**Comment by fdncred** [comment]
Good point. I agree. I'm also curious if closures work for every datatype in regards to theming.

**Comment by Kissaki** [comment]
@fdncred I assume you replied to my comment that I deleted; I noticed that the linked release note has a warning about limited usability/effectiveness - only in specific contexts. I deleted the comment then because I hadn't taken that into account / didn't fully understand it yet.

I created updated documentation in PR #1773 which includes that caveat and my confirmed findings when testing it out.
